playback/src/player.rs

```diff
@@ -2079,10 +2087,7 @@ impl<T: Read + Seek> Seek for Subfile<T> {
         };
 
         let newpos = self.stream.seek(pos)?;
-        if newpos > self.offset {
-            Ok(newpos - self.offset)
-        } else {
-            Ok(0)
-        }
+
+        Ok(newpos.saturating_sub(self.offset))
```

Comment:
This is improved in `new-api` and will cause a merge conflict, but a small one probably. What you wish.

Reply:
I can deal with conflicts.